modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -2226,12 +2240,69 @@ private ObjectEntry _updateObjectEntry(
 				serviceBuilderObjectEntry, scopeKey));
 	}
 
+	private void _updateTaxonomyCategoryIds(
+			long companyId, ObjectEntry objectEntry)
+		throws Exception {
+
+		TaxonomyCategoryBrief[] taxonomyCategoryBriefs =
+			objectEntry.getTaxonomyCategoryBriefs();
+
+		if ((taxonomyCategoryBriefs == null) ||
+			!FeatureFlagManagerUtil.isEnabled("LPD-47858")) {
+
+			return;
+		}
+
+		if (ArrayUtil.isEmpty(taxonomyCategoryBriefs)) {
+			objectEntry.setTaxonomyCategoryIds(() -> new Long[0]);
+		}
+
+		Set<Long> assetCategoryIds = new HashSet<>();
+
+		for (TaxonomyCategoryBrief taxonomyCategoryBrief :
+				taxonomyCategoryBriefs) {
+
+			String externalReferenceCode =
+				taxonomyCategoryBrief.
+					getTaxonomyCategoryExternalReferenceCode();
+
+			Scope scope = taxonomyCategoryBrief.getScope();
+
+			if (Validator.isNull(externalReferenceCode) || (scope == null) ||
+				Validator.isNull(scope.getExternalReferenceCode())) {
+
+				continue;
+			}
+
+			Group group = _groupLocalService.fetchGroupByExternalReferenceCode(
+				scope.getExternalReferenceCode(), companyId);
+
+			if (group == null) {
+				continue;
+			}
+
+			AssetCategory assetCategory =
+				_assetCategoryService.getOrAddIncompleteCategory(
+					externalReferenceCode, group.getGroupId());
+
+			assetCategoryIds.add(assetCategory.getCategoryId());
+		}
+
+		if (SetUtil.isNotEmpty(assetCategoryIds)) {
+			objectEntry.setTaxonomyCategoryIds(
+				() -> assetCategoryIds.toArray(new Long[0]));
+		}
+	}
```

Comment:
@AliciaGarciaGarcia could you please move this logic to [ServiceContextUtil](https://github.com/liferay/liferay-portal/blob/78fb5e057c4b5a5c6c4e20ae1c58f6c6fc20bd90/modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/util/ServiceContextUtil.java#L80-L83), in this place the category ids are set into the service context so the service layer can persist them, other methods different than `addObjectEntry` and `_updateObjectEntry` leverage this method to pass the categories to the service context.

Thanks!